packages/realm/src/app-services/SyncConfiguration.ts

```diff
@@ -366,7 +366,7 @@ function toSSLVerifyCallbackWithListArguments(verifyCallback: SSLVerifyCallback)
 function parseClientResetConfig(clientReset: ClientResetConfig | undefined, onError: ErrorCallback | undefined) {
   if (!clientReset) {
     return {
-      clientResyncMode: undefined,
+      clientResyncMode: toBindingClientResetMode(ClientResetMode.RecoverUnsyncedChanges),
```

Comment:
I would prefer just referring directly to the core enum here:
```suggestion
      clientResyncMode: binding.ClientResetMode.Recover,
```